### PR TITLE
Warn when using @tag outside of describe

### PR DESCRIPTION
**Pull request by sabiwara**
This is a proposal following https://github.com/elixir-lang/elixir/issues/13414 to improve the discoverability.

~~Note: the suite is flaky right now as the test seems to be interfering with other exunit tests, this is just a draft to discuss the proposal.~~

### CODE REVIEW

File: lib/ex_unit/lib/ex_unit/callbacks.ex [hunk 1]
```diff
@@ -765,6 +765,10 @@ defmodule ExUnit.Callbacks do
       raise "@describetag must be set inside describe/2 blocks"
     end
 
+    if Module.get_attribute(module, :tag) != [] do
+      IO.warn("@tag only affects the next test, did you mean to use @describetag?")
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/callbacks.ex:769]:
```suggestion
      IO.warn("found unused @tag before \"describe\", did you mean to use @describetag?")
```

**Review by josevalim — COMMENTED** [review]
LGTM.

File: lib/ex_unit/test/ex_unit/case_test.exs [hunk 2]
```diff
@@ -142,6 +142,23 @@ defmodule ExUnit.CaseTest do
                    end
                  end
   end
+
+  test "warns when using @tag outside of describe" do
+    stderr =
+      ExUnit.CaptureIO.capture_io(:stderr, fn ->
+        defmodule TagOutsideOfDescribe do
+          use ExUnit.Case, register: false
```
Comment (sabiwara) [inline lib/ex_unit/test/ex_unit/case_test.exs:150]: I found this way of fixing test isolation, otherwise would get flaky counts in `ExUnit.RegisterTest`.

Comment (josevalim) [reply lib/ex_unit/test/ex_unit/case_test.exs:150]: You can also raise inside the test, so the module is not defined after all.

**Review by josevalim — APPROVED** [review]

**Pull request merged**